daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/Runner.scala

```diff
@@ -114,6 +108,12 @@ case class Participants[+T](
           .orElse(default_participant)
           .toRight(s"No participant $participant and no default participant")
     }
+
+  def map[A](f: T => A): Participants[A] =
+    copy(
+      default_participant = default_participant.map(f),
+      participants = participants.map { case (k, v) => (k, f(v)) },
```

Comment:
You could also write: 
```suggestion
      participants = participants.transform((_, v) => f(v)),
```

Reply:
Oh thats nice! I was surprised there isn't a nice way to `mapValues`, seems that was removed in pursuit of this `MapView` stuff